core/trino-main/src/main/java/io/trino/metadata/ResolvedFunction.java

```diff
@@ -27,122 +25,36 @@
 import io.trino.spi.type.TypeSignature;
 
 import java.util.Map;
-import java.util.Objects;
 import java.util.Set;
 
 import static com.google.common.base.Preconditions.checkArgument;
 import static java.util.Objects.requireNonNull;
 
-public class ResolvedFunction
+public record ResolvedFunction(
```

Comment:
move to separate PR please

Reply:
https://github.com/trinodb/trino/pull/21593